src/reducers/videoReducer.jsx

```diff
@@ -1,22 +1,34 @@
-export function videoReducer(videoData, action) {
+export function videoReducer(state, action) {
   switch (action.type) {
     case "ALL_CATEGORIES":
-      return { ...videoData, categoryList: action.payload };
+      return { ...state, categoryList: action.payload };
     case "ADD_TO_WATCHLIST":
-      return { ...videoData, watchList: action.payload };
+      return { ...state, watchList: action.payload };
     case "REMOVE_FROM_WATCHLIST":
-      return { ...videoData, watchList: action.payload };
+      return { ...state, watchList: action.payload };
     case "ADD_TO_LIKEDLIST":
-      return { ...videoData, likedList: action.payload };
+      return { ...state, likedList: action.payload };
     case "REMOVE_FROM_LIKEDLIST":
-      return { ...videoData, likedList: action.payload };
+      return { ...state, likedList: action.payload };
     case "ADD_TO_HISTORY":
-      return { ...videoData, history: action.payload };
+      return { ...state, history: action.payload };
     case "DELETE_FROM_HISTORY":
-      return { ...videoData, history: action.payload };
+      return { ...state, history: action.payload };
     case "CLEAR_HISTORY":
-      return { ...videoData, history: action.payload };
+      return { ...state, history: action.payload };
+    case "CREATE_PLAYLIST":
+      return { ...state, playlists: action.payload };
+    case "DELETE_PLAYLIST":
+      return { ...state, playlists: action.payload };
+    case "ADD_TO_PLAYLIST": {
+      let updatedPlaylist = state.playlists.map((item) =>
+        item._id === action.payload._id ? action.payload : item
+      );
+      return { ...state, playlists: updatedPlaylist };
+    }
+    case "DELETE_FROM_PLAYLIST":
+      return state;
     default:
-      return videoData;
+      return state;
```

Comment:
default and DELETE_FROM_PLAYLIST both return the state??

Reply:
I missed changing this, thank you for pointing it out!